### PR TITLE
Refresh recording rules

### PR DESCRIPTION
**Pull request by QuentinBisson**
Signed-off-by: QuentinBisson <quentin@giantswarm.io>

### CODE REVIEW

**Comment by mig4** [comment]
Could you give us a bit of background and explanation please, like what's this for, was it copied or generated from something (seems like a number of unrelated changes, most only formatting?)?

File: helm/prometheus-meta-operator/templates/recording-rules/cortex.rules.yml [hunk 1]
```diff
@@ -10,184 +10,180 @@ spec:
...
+    - expr: sum(rate(nginx_ingress_controller_requests{namespace="giantswarm", ingress="api"}[5m])) by (status)
+      record: aggregation:giantswarm:api_requests
+    - expr: count(app_operator_app_info{status="deployed",namespace="giantswarm"}) by (name,version)
+      record: aggregation:giantswarm:app_deployed_control_plane_total
+    - expr: count(app_operator_app_info{status=~"deployed|DEPLOYED",namespace!="giantswarm"}) by (name,version)
+      record: aggregation:giantswarm:app_deployed_tenant_total
+    - expr: min(cert_exporter_not_after) by (cluster_id, cluster_type)
+      record: aggregation:giantswarm:cluster_certificate_not_after_seconds
+    - expr: label_replace(sum(azure_operator_cluster_release{release_version!=""}) by (release_version, cluster_id), "cluster_id", "$1", "cluster_id", "(.*)") or label_replace(sum(cluster_service_cluster_info) by (release_version, cluster_id) / 2, "cluster_id", "$1", "cluster_id", "(.*)") or label_replace(sum(cluster_operator_cluster_status{release_version!=""}) by (release_version, cluster_id), "cluster_id", "$1", "cluster_id", "(.*)")
```
Comment (TheoBrigitte) [inline helm/prometheus-meta-operator/templates/recording-rules/cortex.rules.yml:44]: Why did this become so complex ?

File: helm/prometheus-meta-operator/templates/recording-rules/cortex.rules.yml [hunk 2]
```diff
@@ -10,184 +10,180 @@ spec:
...
+    - expr: sum(kube_node_status_condition{condition="PIDPressure", status="true"}) by (cluster_type, cluster_id)
+      record: aggregation:kubernetes:node_pid_pressure_total
+    - expr: sum(kube_node_role) by (cluster_type, role, cluster_id)
+      record: aggregation:kubernetes:node_total
+    - expr: sum(kube_persistentvolume_capacity_bytes) by (cluster_type, cluster_id)
+      record: aggregation:kubernetes:persistentvolume_capacity_bytes
+    - expr: count(kube_persistentvolume_info) by (cluster_type, cluster_id)
+      record: aggregation:kubernetes:persistentvolume_total
+    - expr: sum(kube_persistentvolumeclaim_resource_requests_storage_bytes) by (cluster_type, cluster_id)
+      record: aggregation:kubernetes:persistentvolumeclaim_resource_requests_storage_bytes
```
Comment (TheoBrigitte) [inline helm/prometheus-meta-operator/templates/recording-rules/cortex.rules.yml:122]: I don't see this being useful, but it's anyway impossible to tell here whats used and whats not.

File: helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml [hunk 3]
```diff
@@ -10,41 +10,52 @@ spec:
...
   - name: service-level
     rules:
-    - expr: "count(up{app='kubernetes'}) by (cluster_id)"
+    - expr: "count(up{app='kubernetes'}) by (cluster_type,cluster_id)"
       labels:
+        class: HIGH
         service: api-server
       record: raw_slo_requests
-    - expr: "sum((up{app='kubernetes'} * -1) + 1) by (cluster_id)"
+    - expr: "sum((up{app='kubernetes'} * -1) + 1) by (cluster_type,cluster_id)"
```
Comment (TheoBrigitte) [inline helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml:18]: Why do we group by `cluster_type` also now ?

File: helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml [hunk 4]
```diff
@@ -10,41 +10,52 @@ spec:
...
       record: raw_slo_requests
-    - expr: "kube_node_status_condition{condition='Ready', status=~'false|unknown'}"
+    - expr: |
+        (
+          kube_node_status_condition{condition="Ready", status!="true"}
+          and
+          on (node) kube_node_spec_unschedulable == 0
+        )
+        and
+        on (node) time() - kube_node_created > 10 * 60
```
Comment (TheoBrigitte) [inline helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml:35]: Why is that?

**Review by TheoBrigitte — CHANGES_REQUESTED** [review]
Why swapping `expr` and `record` order, it makes review unnecessary hard while nothing changed.
Please make 2 PRs next time. One for moving things around where nothing changes, another one where you do the changes.

I some new changes in here, I guess they come from g8s-prometheus, a raw diff against g8s-prometheus would have been useful I guess.

File: helm/prometheus-meta-operator/templates/recording-rules/cortex.rules.yml (hunk 1 above)
Comment (QuentinBisson) [reply helm/prometheus-meta-operator/templates/recording-rules/cortex.rules.yml:44]: See https://github.com/giantswarm/g8s-prometheus/pull/1209/files

File: helm/prometheus-meta-operator/templates/recording-rules/cortex.rules.yml (hunk 2 above)
Comment (QuentinBisson) [reply helm/prometheus-meta-operator/templates/recording-rules/cortex.rules.yml:122]: I am thinking that once we have all the cortex dashboard as code, we can check which aggregations we use or not

File: helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml (hunk 4 above)
Comment (QuentinBisson) [reply helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml:35]: See https://github.com/giantswarm/g8s-prometheus/pull/1140

File: helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml (hunk 3 above)
Comment (QuentinBisson) [reply helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml:18]: See https://github.com/giantswarm/g8s-prometheus/pull/1150

**Comment by QuentinBisson** [comment]
Just to clarify the situation here as I mentionned that the rules were formatted in another PR, I merely used our transforming tools to regenegate the rules in case there were any changes, which there were.
The tool renders the properties by alphabetical number by default so it seems like it changed a lot and when I initially added the rules, we did not have the tool, which explains the order change.

I would rather not split that in 2 PRs as I would need to do a lot of manual change because our tool does not respect the original property order when rendering the output

**Comment by mig4** [comment]
> Just to clarify the situation here as I mentionned that the rules were formatted in another PR, I merely used our transforming tools to regenegate the rules in case there were any changes, which there were.
> The tool renders the properties by alphabetical number by default so it seems like it changed a lot and when I initially added the rules, we did not have the tool, which explains the order change.

This makes sense to me, if this is mostly a _"copy from g8s-prometheus with some format massaging to make it fit"_ I'm happy, as long as @TheoBrigitte is also happy 😉

**Review by mig4 — APPROVED** [review]

File: helm/prometheus-meta-operator/templates/recording-rules/cortex.rules.yml (hunk 2 above)
Comment (TheoBrigitte) [reply helm/prometheus-meta-operator/templates/recording-rules/cortex.rules.yml:122]: Let's do that then

File: helm/prometheus-meta-operator/templates/recording-rules/cortex.rules.yml (hunk 1 above)
Comment (TheoBrigitte) [reply helm/prometheus-meta-operator/templates/recording-rules/cortex.rules.yml:44]: This does not explain the diff seen here

File: helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml (hunk 3 above)
Comment (TheoBrigitte) [reply helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml:18]: Ok

File: helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml (hunk 4 above)
Comment (TheoBrigitte) [reply helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml:35]: Ok

**Review by TheoBrigitte — APPROVED** [review]
Fine with me, I didn't knew the transformation tool changed.

Maybe next time in the PR description, mention that you used a tool to do this, which command was ran, and a diff against the other repo would help if possible.

File: helm/prometheus-meta-operator/templates/recording-rules/cortex.rules.yml (hunk 1 above)
Comment (QuentinBisson) [reply helm/prometheus-meta-operator/templates/recording-rules/cortex.rules.yml:44]: Actually, yes as:

`
expr: "label_replace(sum(cluster_service_cluster_info) by (release_version, cluster_id) / 2, 'cluster_id', '$1', 'cluster_id', '(.*)') or label_replace(sum(cluster_operator_cluster_status{release_version!=''}) by (release_version, cluster_id), 'cluster_id', '$1', 'cluster_id', '(.*)')"
`

Became: 

`
- expr: label_replace(sum(azure_operator_cluster_release{release_version!=""}) by (release_version, cluster_id), "cluster_id", "$1", "cluster_id", "(.*)") or label_replace(sum(cluster_service_cluster_info) by (release_version, cluster_id) / 2, "cluster_id", "$1", "cluster_id", "(.*)") or label_replace(sum(cluster_operator_cluster_status{release_version!=""}) by (release_version, cluster_id), "cluster_id", "$1", "cluster_id", "(.*)")
`

in the PR I linked

**Comment by QuentinBisson** [comment]
Sure, I will ensure to describe where this is coming from. As I splitted my original PR, I forgot to recomment this one

**Pull request merged**